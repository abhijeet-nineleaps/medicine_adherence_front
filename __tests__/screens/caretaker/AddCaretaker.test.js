import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Addcaretaker from '../../../src/screens/caretaker/AddCaretaker'
Enzyme.configure({adapter: new Adapter()});
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
jest.mock("@react-native-async-storage/async-storage", () => ({
    default: jest.fn(),
  }));

describe('Caretaker Request', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Addcaretaker navigation={undefined} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
    it('test open save button', () => {
    const mockFn = jest.fn();
    const wrapper = shallow(<Addcaretaker search={mockFn} />);
    wrapper.find('#search').simulate('press');
  });
  it('test open save button', () => {
    const mockFn = jest.fn();
    const wrapper = shallow(<Addcaretaker del={mockFn} />);
    wrapper.find('#delete').simulate('press');
  });
});

// describe('Click send image', () => {
//   // it('renders correctly', () => {
//   //   const tree = renderer
//   //     .create(<Addcaretaker navigation={undefined} />)
//   //     .toJSON();
//   //   expect(tree).toMatchSnapshot();
//   // });
//   it('test open save button', () => {
//     const mockFn = jest.fn();
//     const wrapper = shallow(<Addcaretaker search={mockFn} />);
//     wrapper.find('#search').simulate('press');
//   });
//   // it('test open save button', () => {
//   //   const mockFn = jest.fn();
//   //   const wrapper = shallow(<Addcaretaker emp={mockFn} />);
//   //   wrapper.find('#emp').simulate('press');
//   // });
//   it('test open save button', () => {
//     const mockFn = jest.fn();
//     const wrapper = shallow(<Addcaretaker del={mockFn} />);
//     wrapper.find('#delete').simulate('press');
//   });
// });
