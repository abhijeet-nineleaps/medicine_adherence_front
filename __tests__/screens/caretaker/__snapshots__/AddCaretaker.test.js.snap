// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caretaker Request renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "height": "100%",
    }
  }
>
  <RCTScrollView
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "bottom": 0,
            "justifyContent": "flex-end",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "white",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="none"
        style={
          Object {
            "backgroundColor": "white",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            Object {
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "marginBottom": 16,
                "marginRight": 24,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "white",
                    "borderRadius": 5,
                    "color": "black",
                    "elevation": 2,
                    "marginHorizontal": 16,
                    "marginVertical": 8,
                    "paddingHorizontal": 12,
                    "paddingVertical": 6,
                  },
                  undefined,
                ]
              }
            >
              Add Caretaker
            </Text>
            <View
              collapsable={false}
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 28,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "center",
                  "opacity": 1,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 3,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 3,
                    },
                    Object {
                      "borderRadius": 28,
                      "elevation": 4,
                    },
                    false,
                  ]
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "busy": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#3743ab",
                        "borderColor": "#2089dc",
                        "borderRadius": 3,
                        "borderWidth": 0,
                        "flexDirection": "row",
                        "height": 40,
                        "justifyContent": "center",
                        "padding": 8,
                        "width": 40,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "height": 40,
                          "marginHorizontal": 5,
                          "overflow": "hidden",
                          "padding": 8,
                          "width": 40,
                        }
                      }
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": "white",
                                  "fontSize": 24,
                                },
                                Object {
                                  "backgroundColor": "transparent",
                                },
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                            testID="iconIcon"
                          >
                            î……
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            Object {
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "marginBottom": 16,
                "marginRight": 24,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "white",
                    "borderRadius": 5,
                    "color": "black",
                    "elevation": 2,
                    "marginHorizontal": 16,
                    "marginVertical": 8,
                    "paddingHorizontal": 12,
                    "paddingVertical": 6,
                  },
                  undefined,
                ]
              }
            >
              Delete
            </Text>
            <View
              collapsable={false}
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 28,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "center",
                  "opacity": 1,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 3,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 3,
                    },
                    Object {
                      "borderRadius": 28,
                      "elevation": 4,
                    },
                    false,
                  ]
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "busy": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#3743ab",
                        "borderColor": "#2089dc",
                        "borderRadius": 3,
                        "borderWidth": 0,
                        "flexDirection": "row",
                        "height": 40,
                        "justifyContent": "center",
                        "padding": 8,
                        "width": 40,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "height": 40,
                          "marginHorizontal": 5,
                          "overflow": "hidden",
                          "padding": 8,
                          "width": 40,
                        }
                      }
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": "white",
                                  "fontSize": 24,
                                },
                                Object {
                                  "backgroundColor": "transparent",
                                },
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                            testID="iconIcon"
                          >
                            î¡²
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 28,
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": 16,
              "marginTop": 0,
              "opacity": 1,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 3,
                },
                Object {
                  "borderRadius": 28,
                  "elevation": 4,
                },
                false,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#3743ab",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "height": 56,
                    "justifyContent": "center",
                    "padding": 16,
                    "width": 56,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 56,
                      "marginHorizontal": 5,
                      "overflow": "hidden",
                      "padding": 16,
                      "width": 56,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 24,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        î……
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTSafeAreaView>
    </View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            A
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
